manifests/config/lookup_tables.pp

```diff
@@ -0,0 +1,10 @@
+class rsyslog::config::lookup_tables {
+  $::rsyslog::config::lookup_tables.each |$table, $config| {
+    rsyslog::component::lookup_table { $table:
+      * => {
```

Comment:
Here I actually am creating the parameters dynamically. When `rsyslog::config::lookup_tables` is parsed into `$table` and `$config`, `$config` is a nested hash (that also has another nested hash, `lookup_json`) inside of it.

Now, I could be missing a better way to do this as I just read the documentation on the advanced resource expressions today.